crates/ruff_linter/src/rules/eradicate/detection.rs

```diff
@@ -1,43 +1,45 @@
 /// See: [eradicate.py](https://github.com/myint/eradicate/blob/98f199940979c94447a461d50d27862b118b282d/eradicate.py)
+use aho_corasick::AhoCorasick;
 use once_cell::sync::Lazy;
 use regex::Regex;
 
 use ruff_python_parser::parse_suite;
 
+static CODE_INDICATOR: Lazy<AhoCorasick> = Lazy::new(|| {
+    AhoCorasick::new([
+        "(", ")", "[", "]", "{", "}", ":", "=", "%", "print", "return", "break", "continue",
+        "import",
+    ])
+    .unwrap()
+});
 static ALLOWLIST_REGEX: Lazy<Regex> = Lazy::new(|| {
     Regex::new(
         r"^(?i)(?:pylint|pyright|noqa|nosec|region|endregion|type:\s*ignore|fmt:\s*(on|off)|isort:\s*(on|off|skip|skip_file|split|dont-add-imports(:\s*\[.*?])?)|mypy:|SPDX-License-Identifier:)",
     ).unwrap()
 });
 static BRACKET_REGEX: Lazy<Regex> = Lazy::new(|| Regex::new(r"^[()\[\]{}\s]+$").unwrap());
-static CODE_INDICATORS: &[&str] = &[
-    "(", ")", "[", "]", "{", "}", ":", "=", "%", "print", "return", "break", "continue", "import",
-];
-static CODE_KEYWORDS: Lazy<Vec<Regex>> = Lazy::new(|| {
-    vec![
-        Regex::new(r"^\s*elif\s+.*\s*:\s*$").unwrap(),
-        Regex::new(r"^\s*else\s*:\s*$").unwrap(),
-        Regex::new(r"^\s*try\s*:\s*$").unwrap(),
-        Regex::new(r"^\s*finally\s*:\s*$").unwrap(),
-        Regex::new(r"^\s*except\s+.*\s*:\s*$").unwrap(),
-    ]
+static CODE_KEYWORDS: Lazy<Regex> = Lazy::new(|| {
+    Regex::new(r"^\s*(?:elif\s+.*\s*:|else\s*:|try\s*:|finally\s*:|except\s+.*\s*:)\s*$").unwrap()
 });
 static CODING_COMMENT_REGEX: Lazy<Regex> =
     Lazy::new(|| Regex::new(r"^.*?coding[:=][ \t]*([-_.a-zA-Z0-9]+)").unwrap());
 static HASH_NUMBER: Lazy<Regex> = Lazy::new(|| Regex::new(r"#\d").unwrap());
-static MULTILINE_ASSIGNMENT_REGEX: Lazy<Regex> =
-    Lazy::new(|| Regex::new(r"^\s*([(\[]\s*)?(\w+\s*,\s*)*\w+\s*([)\]]\s*)?=.*[(\[{]$").unwrap());
+static MULTILINE_ASSIGNMENT_REGEX: Lazy<Regex> = Lazy::new(|| {
+    Regex::new(r"^\s*(?:[(\[]\s*)?(?:\w+\s*,\s*)*\w+\s*([)\]]\s*)?=.*[(\[{]$").unwrap()
+});
 static PARTIAL_DICTIONARY_REGEX: Lazy<Regex> =
     Lazy::new(|| Regex::new(r#"^\s*['"]\w+['"]\s*:.+[,{]\s*(#.*)?$"#).unwrap());
-static PRINT_RETURN_REGEX: Lazy<Regex> = Lazy::new(|| Regex::new(r"^(print|return)\b\s*").unwrap());
+static PRINT_RETURN_REGEX: Lazy<Regex> =
+    Lazy::new(|| Regex::new(r"^(?:print|return)\b\s*").unwrap());
 
 /// Returns `true` if a comment contains Python code.
 pub(crate) fn comment_contains_code(line: &str, task_tags: &[String]) -> bool {
-    let line = if let Some(line) = line.trim().strip_prefix('#') {
-        line.trim_start_matches([' ', '#'])
-    } else {
+    let line = line.trim_start_matches([' ', '#']).trim_end();
+
+    // Fast path: if none of the indicators are present, the line is not code.
+    if !CODE_INDICATOR.is_match(line) {
         return false;
-    };
+    }
```

Comment:
Put the cheapest and most discriminant step first. This should exclude most comments.